### DISCUSSION TITLE
Unknown Error: 'tracker'

### DISCUSSION THREAD

**Discussion by b0ril** [StuffAnThings/qbit_manage#238] [Q&A] [answered]
Hello,
I'm running docker instance of qbit_manage with very simple config file and I'm getting this error:

====================================================================================================|
|                                            Starting Run                                            |
|====================================================================================================|
| Locating config...                                                                                 |
| Using /config/config.yml as config                                                                 |
| Commands found in config.yml, ignoring env variables and using config commands instead.            |
| Connecting to Qbittorrent...                                                                       |
| Qbt Connection Successful                                                                          |
|======================================= Getting Torrent List =======================================|
|======================================== Checking Settings =========================================|
|====================================================================================================|
|                                   Gathering Torrent Information                                    |
|====================================================================================================|
|=============================== Tagging Torrents with Tracker Errors ===============================|
| Unknown Error: 'tracker'                                                                           |
| Unknown Error: 'tracker'                                                                           |
| No unregistered torrents found.                                                                    |
|==================================== Rechecking Paused Torrents ====================================|
|====================================================================================================|
|                                            Finished Run                                            |
|                                         Run Time: 0:00:00                                          |
|                    Current Time: 00:44 | 2 Minutes until the next run at 00:46                     |
|====================================================================================================

Here is my config file:

commands:
  # The commands defined below will IGNORE any commands used in command line and docker env variables.
  dry_run: false
  cross_seed: false
  recheck: true
  cat_update: false
  tag_update: false
  rem_unregistered: true
  tag_tracker_error: true
  rem_orphaned: false
  tag_nohardlinks: false
  skip_cleanup: false

qbt:
  # qBittorrent parameters
  host: 192.168.1.100:8080
  user: admin
  pass: ********

settings:
  force_auto_tmm: false  # Will force qBittorrent to enable Automatic Torrent Management for each torrent.
  tracker_error_tag: issue  # Will set the tag of any torrents that do not have a working tracker.
  ignoreTags_OnUpdate: # When running tag-update function, it will update torrent tags for a given torrent even if the torrent has one or more of the tags defined here.
  - noHL
  - issue
  - cross-seed
directory:
  # Do not remove these
  # Cross-seed var: </your/path/here/>  # Output directory of cross-seed
  # root_dir var: </your/path/here/>  # Root downloads directory used to check for orphaned files, noHL, and RecycleBin.
  # <OPTIONAL> remote_dir var: </your/path/here/>  # Path of docker host mapping of root_dir.
  # Must be set if you're running qbit_manage locally and qBittorrent/cross_seed is in a docker
  # <OPTIONAL> recycle_bin var: </your/path/here/>   # Path of the RecycleBin folder. Default location is set to remote_dir/.RecycleBin
  # <OPTIONAL> torrents_dir var: </your/path/here/>  # Path of the your qbittorrent torrents directory. Required for `save_torrents` attribute in recyclebin

  cross_seed: /downloads/incomplete
  root_dir: /downloads
  #remote_dir: "/mnt/user/data/torrents/"                                                                                                                   
  #recycle_bin: "/mnt/user/data/torrents/.RecycleBin"                                                                                                       
  #torrents_dir: "/qbittorrent/data/BT_backup"                                                                                                              
  #orphaned_dir: "/data/torrents/orphaned_data"                                                                                                             
  remote_dir: /downloads/
  torrents_dir: /torrents

movies:
    # <OPTIONAL> exclude_tags var: Will exclude the following tags when searching through the category.
    exclude_tags:
    # <OPTIONAL> cleanup var: WARNING!! Setting this as true Will remove and delete contents of any torrents that are in paused state and has the NoHL tag
    cleanup: true
    # <OPTIONAL> max_ratio var: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading
    max_ratio: 2.0
    # <OPTIONAL> max seeding time var: Will set the torrent Maximum seeding time (min) until torrent is stopped from seeding
    max_seeding_time: 14400
    # <OPTIONAL> Limit Upload Speed var: Will limit the upload speed KiB/s (KiloBytes/second) (`-1` : No Limit)
    limit_upload_speed:
    # <OPTIONAL> min seeding time var: Will ensure that torrent is not deleted by cleanup variable if torrent does not meet minimum seeding time (min).
    min_seeding_time: 14400

  # Can have additional categories set with separate ratio/seeding times defined.
tv:
    # <OPTIONAL> exclude_tags var: Will exclude the following tags when searching through the category.
    exclude_tags:
    # <OPTIONAL> cleanup var: WARNING!! Setting this as true Will remove and delete contents of any torrents that are in paused state and has the NoHL tag
    cleanup: true
    # <OPTIONAL> max_ratio var: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading
    max_ratio: 2.0
    # <OPTIONAL> max seeding time var: Will set the torrent Maximum seeding time (min) until torrent is stopped from seeding
    max_seeding_time: 14400
    # <OPTIONAL> Limit Upload Speed var: Will limit the upload speed KiB/s (KiloBytes/second) (`-1` : No Limit)
    limit_upload_speed:
    # <OPTIONAL> min seeding time var: Will ensure that torrent is not deleted by cleanup variable if torrent does not meet minimum seeding time (min).
    min_seeding_time: 14400

recyclebin:
  # Recycle Bin method of deletion will move files into the recycle bin (Located in /root_dir/.RecycleBin) instead of directly deleting them in qbit
  # By default the Recycle Bin will be emptied on every run of the qbit_manage script if empty_after_x_days is defined.
  enabled: false
  # <OPTIONAL> empty_after_x_days var: 
  # Will automatically remove all files and folders in recycle bin after x days. (Checks every script run)
  # If this variable is not defined it, the RecycleBin will never be emptied.
  # WARNING: Setting this variable to 0 will delete all files immediately upon script run!
  empty_after_x_days: 60
  # <OPTIONAL> save_torrents var:
  # If this option is set to true you MUST fill out the torrents_dir in the directory attribute.
  # This will save a copy of your .torrent and .fastresume file in the recycle bin before deleting it from qbittorrent
  save_torrents: false
  # <OPTIONAL> split_by_category var:
  # This will split the recycle bin folder by the save path defined in the `cat` attribute 
  # and add the base folder name of the recycle bin that was defined in the `recycle_bin` sub-attribute under directory.
  split_by_category: false

**Comment by bakerboy448** [comment, marked as answer]
your config file is invalid; 
- you have no trackers
- you have categories randomly set in the file for some reason and not under the nohardlinks header

follow the config file example and see the wiki and configure your config file as required; you can't just randomly yeet things from the config file.

**Reply by b0ril** [reply]
I don't want to use trackers.
I want just to remove unregistered torrents and when ratio or time is reached ?
I'm not using hardlinks for my torrents

**Reply by bakerboy448** [reply]
then you can't use qbm - you cannot arbitrary decide to not use required config file items. 

Hardlinks have nothing to do with trackers, no idea what you're talking about.....but also no idea why you have categories randomly configured in your configuration file either.

You can't just cherry pick and make your own config file ignoring required components.

https://github.com/StuffAnThings/qbit_manage/wiki/Config-Setup

**Comment by b0ril** [comment]
Thank you for your prompt support.
I figured out that adding below option

tracker:                                                                                                                                                    
   other:                                                                                                                                                   
    tag: other 

fix my setup
now it is working perfectly fine

**Reply by b0ril** [reply]
`# This is an example configuration file that documents all the options.
# It will need to be modified for your specific use case.
# Please refer to the link below for more details on how to set up the configuration file
# https://github.com/StuffAnThings/qbit_manage/wiki/Config-Setup

commands:
  # The commands defined below will IGNORE any commands used in command line and docker env variables.
  dry_run: false
  cross_seed: false
  recheck: true
  cat_update: false
  tag_update: false
  rem_unregistered: true
  tag_tracker_error: true
  rem_orphaned: false
  tag_nohardlinks: false
  skip_cleanup: false

qbt:
  # qBittorrent parameters
  host: 192.168.1.100:8080
  user: admin
  pass: ******

settings:
  force_auto_tmm: false  # Will force qBittorrent to enable Automatic Torrent Management for each torrent.
  tracker_error_tag: issue  # Will set the tag of any torrents that do not have a working tracker.
  ignoreTags_OnUpdate: # When running tag-update function, it will update torrent tags for a given torrent even if the torrent has one or more of the tags defined here.
  - noHL
  - issue
  - cross-seed
directory:
  # Do not remove these
  # Cross-seed var: </your/path/here/>  # Output directory of cross-seed
  # root_dir var: </your/path/here/>  # Root downloads directory used to check for orphaned files, noHL, and RecycleBin.
  # <OPTIONAL> remote_dir var: </your/path/here/>  # Path of docker host mapping of root_dir.
  # Must be set if you're running qbit_manage locally and qBittorrent/cross_seed is in a docker
  # <OPTIONAL> recycle_bin var: </your/path/here/>   # Path of the RecycleBin folder. Default location is set to remote_dir/.RecycleBin
  # <OPTIONAL> torrents_dir var: </your/path/here/>  # Path of the your qbittorrent torrents directory. Required for `save_torrents` attribute in recyclebin

  cross_seed: /downloads/incomplete
  root_dir: /downloads
  #remote_dir: "/mnt/user/data/torrents/"                                                                                                                   
  #recycle_bin: "/mnt/user/data/torrents/.RecycleBin"                                                                                                       
  #torrents_dir: "/qbittorrent/data/BT_backup"                                                                                                              
  #orphaned_dir: "/data/torrents/orphaned_data"                                                                                                             
  remote_dir: /downloads/
  torrents_dir: /torrents

tracker:
   other:
    tag: other

#nohardlinks:
  # Tag Movies/Series that are not hard linked
  # Mandatory to fill out directory parameter above to use this function (root_dir/remote_dir)
  # This variable should be set to your category name of your completed movies/completed series in qbit. Acceptable variable can be any category you would like to tag if there are no hardlinks found
movies:
    # <OPTIONAL> exclude_tags var: Will exclude the following tags when searching through the category.
    exclude_tags:
    # <OPTIONAL> cleanup var: WARNING!! Setting this as true Will remove and delete contents of any torrents that are in paused state and has the NoHL tag
    cleanup: true
    # <OPTIONAL> max_ratio var: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading
    max_ratio: 2.0
    # <OPTIONAL> max seeding time var: Will set the torrent Maximum seeding time (min) until torrent is stopped from seeding
    max_seeding_time: 14400
    # <OPTIONAL> Limit Upload Speed var: Will limit the upload speed KiB/s (KiloBytes/second) (`-1` : No Limit)
    limit_upload_speed:
    # <OPTIONAL> min seeding time var: Will ensure that torrent is not deleted by cleanup variable if torrent does not meet minimum seeding time (min).
    min_seeding_time: 14400

  # Can have additional categories set with separate ratio/seeding times defined.
tv:
    # <OPTIONAL> exclude_tags var: Will exclude the following tags when searching through the category.
    exclude_tags:
    # <OPTIONAL> cleanup var: WARNING!! Setting this as true Will remove and delete contents of any torrents that are in paused state and has the NoHL tag
    cleanup: true
    # <OPTIONAL> max_ratio var: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading
    max_ratio: 2.0
    # <OPTIONAL> max seeding time var: Will set the torrent Maximum seeding time (min) until torrent is stopped from seeding
    max_seeding_time: 14400
    # <OPTIONAL> Limit Upload Speed var: Will limit the upload speed KiB/s (KiloBytes/second) (`-1` : No Limit)
    limit_upload_speed:
    # <OPTIONAL> min seeding time var: Will ensure that torrent is not deleted by cleanup variable if torrent does not meet minimum seeding time (min).
    min_seeding_time: 14400

recyclebin:
  # Recycle Bin method of deletion will move files into the recycle bin (Located in /root_dir/.RecycleBin) instead of directly deleting them in qbit
  # By default the Recycle Bin will be emptied on every run of the qbit_manage script if empty_after_x_days is defined.
  enabled: false
  # <OPTIONAL> empty_after_x_days var: 
  # Will automatically remove all files and folders in recycle bin after x days. (Checks every script run)
  # If this variable is not defined it, the RecycleBin will never be emptied.
  # WARNING: Setting this variable to 0 will delete all files immediately upon script run!
  empty_after_x_days: 60
  # <OPTIONAL> save_torrents var:
  # If this option is set to true you MUST fill out the torrents_dir in the directory attribute.
  # This will save a copy of your .torrent and .fastresume file in the recycle bin before deleting it from qbittorrent
  save_torrents: true
  # <OPTIONAL> split_by_category var:
  # This will split the recycle bin folder by the save path defined in the `cat` attribute 
  # and add the base folder name of the recycle bin that was defined in the `recycle_bin` sub-attribute under directory.
  split_by_category: false

orphaned:
  # Orphaned files are those in the root_dir download directory that are not referenced by any active torrents.
  # Will automatically remove all files and folders in orphaned data after x days. (Checks every script run)
  # If this variable is not defined it, the orphaned data will never be emptied.
  # WARNING: Setting this variable to 0 will delete all files immediately upon script run!
  empty_after_x_days: 7
  # File patterns that will not be considered orphaned files. Handy for generated files that aren't part of the torrent but belong with the torrent's files
  exclude_patterns:
webhooks:
  error:
  run_start:
  run_end:
  function:
    cross_seed:
    recheck:
    cat_update:
    tag_update:
    rem_unregistered:
    tag_tracker_error:
    rem_orphaned:
    tag_nohardlinks:
    cleanup_dirs:`

**Reply by bakerboy448** [reply]
Confirming for anyone landing here via search: your fix is correct and still valid — a `tracker:` section with `other: tag: other` (or any keyword) is unchanged in the current config schema.

Root cause: qbm requires at least one of `cat` or `tracker` to be non-empty. With both empty, it hit an unhandled `KeyError('tracker')` deep in tracker-tagging logic, surfaced generically as `Unknown Error: 'tracker'`.

As of v4.5.0 this fails fast at config-load time instead, with an explicit message: *"Config Error: Both 'cat' (categories) and 'tracker' sections are empty. At least one must be defined and contain valid entries."* (`modules/config.py`, `validate_required_sections`). So on current versions you'd get a clear error immediately rather than the cryptic `Unknown Error: 'tracker'`.
